apps/tx-builder/src/lib/storage.ts

```diff
@@ -69,14 +69,24 @@ const getBatches = async () => {
 }
 
 const downloadObjectAsJson = (batchFile: BatchFile) => {
-  const dataStr =
-    'data:text/json;charset=utf-8,' +
-    encodeURIComponent(JSON.stringify(batchFile, stringifyReplacer))
+  const blobURL = URL.createObjectURL(
+    new Blob([JSON.stringify(batchFile, stringifyReplacer)], { type: 'application/json' }),
+  )
+
+  // If Firefox or Safari open a new window to download the file
+  // https://bugzilla.mozilla.org/show_bug.cgi?id=1365502
+  if (
+    navigator.userAgent.indexOf('Firefox') !== -1 ||
+    (navigator.userAgent.indexOf('Safari') !== -1 && navigator.userAgent.indexOf('Chrome') === -1)
```

Comment:
What about using `includes()` that already returns a `boolean` value? Also this is case sensitive

https://www.w3schools.com/jsref/jsref_includes.asp